Schema/PropertyIgnoreContractResolver.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Schema
+{
+    public class PropertyIgnoreContractResolver : DefaultContractResolver
+    {
+        private readonly Dictionary<Type, HashSet<string>> _ignores;
+
+        public PropertyIgnoreContractResolver()
+        {
+            _ignores = new Dictionary<Type, HashSet<string>>();
+        }
+
+        public void IgnoreProperty(Type type, params string[] jsonPropertyNames)
+        {
+            if (!_ignores.ContainsKey(type))
+            {
+                _ignores[type] = new HashSet<string>();
+            }
+
+            foreach (var prop in jsonPropertyNames)
+            {
+                _ignores[type].Add(prop);
+            }
+        }
+
+        protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
+        {
+            var property = base.CreateProperty(member, memberSerialization);
+
+            if (IsIgnored(property.DeclaringType, property.PropertyName))
+            {
+                property.ShouldSerialize = i => false;
+                property.Ignored = true;
+            }
+
+            return property;
+        }
+
+        private bool IsIgnored(Type type, string jsonPropertyName)
+        {
+            if (!_ignores.ContainsKey(type))
```

Comment:
Not thread-safe if that's an issue

Reply:
this would always be single-threaded and only needs to be run when there are schema changes for Steeltoe config keys